index.js

```diff
@@ -32,24 +33,32 @@ exports.register = (server, pluginOptions, next) => {
   // eg: default-src https: 'unsafe-inline' 'unsafe-eval'; report-uri https://example.com/reportingEndpoint
   const cspValue = Object.keys(options.fetchDirectives).reduce((memo, fetchDirective) => {
     const fetchDirectiveValue = options.fetchDirectives[fetchDirective];
-    // policy could be either a single policy or list of them:
+    // policy could be either a single policy or list of one or more of them:
     if (typeof fetchDirectiveValue === 'string') {
       memo.push(`${fetchDirective} ${quotify(fetchDirectiveValue)}`);
-    } else {
+    } else if (fetchDirectiveValue.length !== 0) {
       memo.push(`${fetchDirective} ${fetchDirectiveValue.map(item => quotify(item)).join(' ')}`);
     }
     return memo;
   }, []).join(';');
 
   // calculates and adds the CSP header for each request before it returns
   server.ext('onPreResponse', (request, reply) => {
-    // don't worry about it if this was called by the CSP report route:
-    if (options.fetchDirectives['report-uri'] === request.path) {
-      return reply.continue();
-    }
-    // don't worry about it if this response variety isn't in the indicated varieties:
-    if (options.varietiesToInclude.indexOf(request.response.variety) < 0) {
-      return reply.continue();
+    // unless the cspHeader option is set for this route, check if we still need to intercept the response:
+    if (!request.route.settings.plugins['hapi-csp-mixed'] || !request.route.settings.plugins['hapi-csp-mixed'].cspHeaders) {
```

Comment:
should check plugin cspHeaders != true or varieties !include and then return reply.contine().  This should be the second check.